sds_data_manager/stacks/ialirt_ingest_lambda_stack.py

```diff
@@ -0,0 +1,124 @@
+"""Configure the ialirt ingest lambda stack."""
+
+import pathlib
+
+import aws_cdk as cdk
+from aws_cdk import Stack, aws_dynamodb, aws_s3
+from aws_cdk import aws_events as events
+from aws_cdk import aws_events_targets as targets
+from aws_cdk import aws_iam as iam
+from aws_cdk import aws_lambda as lambda_
+from aws_cdk import aws_lambda_python_alpha as lambda_alpha_
+from constructs import Construct
+
+
+class IalirtIngestLambda(Stack):
+    """Stack for ialirt ingest lambda."""
+
+    def __init__(
+        self,
+        scope: Construct,
+        construct_id: str,
+        env: cdk.Environment,
+        packet_data_table: aws_dynamodb.Table,
+        ialirt_bucket: aws_s3.Bucket,
+        **kwargs,
+    ) -> None:
+        """IalirtIngestLambda Stack.
+
+        Parameters
+        ----------
+        scope : Construct
+            Parent construct.
+        construct_id : str
+            A unique string identifier for this construct.
+        env : obj
+            Environment.
+        packet_data_table : aws_dynamodb.Table
+            Database table.
+        ialirt_bucket : aws_s3.Bucket
+            The data bucket.
+        kwargs : dict
+            Keyword arguments.
+
+        """
+        super().__init__(scope, construct_id, env=env, **kwargs)
+
+        lambda_role = iam.Role(
+            self,
+            "IalirtIngestLambdaRole",
+            assumed_by=iam.ServicePrincipal("lambda.amazonaws.com"),
+            managed_policies=[
+                iam.ManagedPolicy.from_aws_managed_policy_name(
+                    "service-role/AWSLambdaBasicExecutionRole"
+                )
+            ],
+        )
+
+        lambda_role.add_to_policy(
+            iam.PolicyStatement(
+                actions=[
+                    "dynamodb:PutItem",
+                    "dynamodb:BatchGetItem",
+                    "dynamodb:DescribeStream",
+                    "dynamodb:DescribeTable",
+                    "dynamodb:GetItem",
+                    "dynamodb:ListStreams",
+                    "dynamodb:ListTables",
+                    "dynamodb:Query",
+                    "dynamodb:Scan",
+                    "dynamodb:UpdateItem",
+                    "dynamodb:DeleteItem",
+                    "s3:GetObject",
+                ],
+                resources=[
+                    packet_data_table.table_arn,
+                    f"{ialirt_bucket.bucket_arn}/*",
+                ],
+            )
+        )
+
+        ialirt_ingest_lambda = lambda_alpha_.PythonFunction(
+            self,
+            id="IalirtIngestLambda",
+            function_name="ialirt-ingest",
+            entry=str(
+                pathlib.Path(__file__).parent.joinpath("..", "lambda_code").resolve()
+            ),
+            index="IAlirtCode/ialirt_ingest.py",
+            handler="lambda_handler",
+            runtime=lambda_.Runtime.PYTHON_3_9,
```

Comment:
Please bump this to 3_12, we don't want to have to upgrade off an old runtime soon after launch.